MatrixSDK/Crypto/Recovery/MXRecoveryService.m

```diff
@@ -557,26 +555,8 @@ - (void)recoverSecrets:(nullable NSArray<NSString*>*)secrets
             
             NSString *secret = unpaddedBase64Secret;
             
-            // Validate the secret before storing it
-            if ([self checkSecret:secret withSecretId:secretId])
-            {
-                if (![secret isEqualToString:[self.dependencies.secretStore secretWithSecretId:secretId]])
-                {
-                    MXLogDebug(@"[MXRecoveryService] recoverSecrets: Recovered secret %@", secretId);
-                    
-                    [updatedSecrets addObject:secretId];
-                    [self.dependencies.secretStore storeSecret:secret withSecretId:secretId];
-                }
-                else
-                {
-                    MXLogDebug(@"[MXRecoveryService] recoverSecrets: Secret %@ was already known", secretId);
-                }
-            }
-            else
-            {
-                MXLogDebug(@"[MXRecoveryService] recoverSecrets: Secret %@ is invalid", secretId);
-                [invalidSecrets addObject:secretId];
-            }
```

Comment:
No idea, let's wait for someone from Crypto to confirm this

Reply:
Ok, so per my understanding here we are trying to import secrets from the 4S (cross signing keys and megolm backup keys).
- For cross-signing keys, it is supposed to call directly olmMachine.importCrossSigningKeys that will error if secret does not match, and will only store the secret in "cache" if it does.
So by removing the check here it will then change a bit the behavior as the following call to importCrossSigningKeys could then error.

- For backup it is a bit different as all the logic is not in the rust crypto sdk, but still in the native crate. So it appears to me that now **you are not saving the backup decryption key in cache** so the recovery from backup will fail now. I would expect some test to fail here. I would expect the code here to check the validity of the key (without trying to import keys as it would be too long), then to store the secret if valid. Eventually a restore from background could be started

Also looks like you are adding all secrets to `invalidSecrets`, won't this impact/break the UI?

Reply:
okay by investigating further, the problem is that this required `- (BOOL)checkSecret:(NSString*)secret withSecretId:(NSString*)secret` that I have removed, and that I could also restore, but relies also on the `MXCrossSigningTools ` object specifically the `isSecretValid:secret forPublicKeys:crossSigningInfo.masterKeys.keys` function, which is heavily using OLMKit.